src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -37,10 +41,10 @@
                                   TelemetryClient telemetryClient = null,
                                   int chunkSize = 8192)
         {
-            s_telemetryConfiguration = telemetryConfiguration;
-            s_telemetryClient = telemetryClient;
-            s_sessionId = TelemetryEnabled ? Guid.NewGuid().ToString() : null;
-            ChunkSize = chunkSize;
+            s_telemetryConfiguration ??= telemetryConfiguration;
```

Comment:
>telemetryConfiguration [](http://example.com/codeflow?start=41&length=22)

we should only use telemetryConfiguration/telemetryClient if the respective static value is null. This should only be used during tests.